sfall/LoadGameHook.cpp

```diff
@@ -276,6 +277,45 @@ static void __declspec(naked) LoadGame() {
 		retn;
 	}
 }
+
+void ReadExtraGameMsgFiles()
+{
+	int read;
+	std::string names;
+
+	names.resize(256);
+
+	while ((read = GetPrivateProfileStringA("Misc", "ExtraGameMsgFileList", "",
+		(LPSTR)names.data(), names.size(), ".\\ddraw.ini")) == names.size() - 1)
+		names.resize(names.size() + 256);
+
+	if (names.empty())
+		return;
+
+	names.resize(names.find_first_of('\0'));
+	names.append(",");
+
+	int begin = 0;
+	int end;
+	int length;
+
+	while ((end = names.find_first_of(',', begin)) != std::string::npos)
+	{
+		length = end - begin;
+
+		if (length > 0)
+		{
+			std::string path = "game\\" + names.substr(begin, length) + ".msg";
```

Comment:
The new build still doesn't work, but my binary works in your game. Looks like something causing problems in my game/system. There is a similar case that one of the options in hi-res patch conflicts with some sfall functions, making them not work as they should. But this time I can't think of any of it would conflict with this feature. :\